pkg/git/libgit2/checkout.go

```diff
@@ -53,12 +53,34 @@ func CheckoutStrategyForRef(ref *sourcev1.GitRepositoryRef, opt git.CheckoutOpti
 	}
 }
 
+// clone is a wrapper around git2go.Clone that respects the provided context.
+func clone(ctx context.Context, path, url string, auth *git.Auth, opts *git2go.CloneOptions) (*git2go.Repository, error) {
+	var repo *git2go.Repository
+	errCh := make(chan error, 1)
+	go func() {
+		var err error
+		repo, err = git2go.Clone(url, path, opts)
+		errCh <- err
+	}()
+
+	select {
+	case err := <-errCh:
+		if err != nil {
+			return nil, fmt.Errorf("unable to clone '%s', error: %w", url, gitutil.LibGit2Error(err))
```

Comment:
While we're here: it would be more usual to have

```suggestion
			return nil, fmt.Errorf("unable to clone '%s': %w", url, gitutil.LibGit2Error(err))
```